public/app/plugins/panel/table/module.tsx

```diff
@@ -79,6 +80,13 @@ export const plugin = new PanelPlugin<PanelOptions, PanelFieldConfig>(TablePanel
           name: 'Column filter',
           description: 'Enables/disables field filters in table',
           defaultValue: defaultPanelFieldConfig.filterable,
+        })
+        .addBooleanSwitch({
+          path: 'hidden',
+          name: 'Hidden field',
+          description: 'Do not display the column',
```

Comment:
suggestions for better names/text?

Reply:
* Column visibility 
* Field visibility 

then it could be made into radio button with "visible | hidden" options. But not sure this is better :shrug:

Reply:
* `Hide in table` (to be consistent with Hide in area for graph panels) with a boolean switch

Reply:
Updated to 
![image](https://user-images.githubusercontent.com/705951/140957128-8e537a0c-2a24-4c0c-b53b-37f41c8523eb.png)